nx-os/poap/poap.py

```diff
@@ -275,14 +275,15 @@ def abort_cleanup_exit () :
 import signal
 import string
 
-# Set CDP variables for location option
-# Will be used by set_config_file_src_location() function
-poap_log("INFO: show cdp neighbors interface %s" % cdp_interface)
-a = clid("show cdp neighbors interface %s" % cdp_interface)
-b = json.loads(a)
-cdpnei_switchName = str(b['TABLE_cdp_neighbor_brief_info']['ROW_cdp_neighbor_brief_info']['device_id'])
-cdpnei_intfName = str(b['TABLE_cdp_neighbor_brief_info']['ROW_cdp_neighbor_brief_info']['port_id'])
-cdpnei_intfName = string.replace(cdpnei_intfName, "/", "_")
+if config_file_type == "location" and cdp_interface is not None:
```

Comment:
There is a good reason to leave this here though.  It serves the purpose of defensive programming.  If for some reason the lines above this change, move or otherwise get modified to the point that the script does not exit like it used to and we end up here with `config_file_type == 'location'` being `True` we don't end up with another latent bug.  I would rather we only run the `clid` when we have to and when we have the information we need to run it than when we think we have to but don't have the information we need and end up in the same situation regardless of the cause.

You do bring up a good point though, why have this if we don't exit with the same error.  I'll add that and put a comment about defensive programming so others don't remove it to be more efficient.

Thanks!